ospfd/ospf_vty.c

```diff
@@ -677,8 +679,15 @@ DEFUN (ospf_area_range_cost,
 	ospf_area_display_format_set(ospf, ospf_area_get(ospf, area_id),
 				     format);
 
-	cost = strtoul(argv[idx_cost]->arg, NULL, 10);
-	ospf_area_range_cost_set(ospf, area_id, &p, cost);
+	if (argv_find(argv, argc, "cost", &idx)) {
+		cost = strtoul(argv[idx+1]->arg, NULL, 10);
+		ospf_area_range_cost_set(ospf, area_id, &p, cost);
+	}
+
+	if (argv_find(argv, argc, "substitute", &idx)) {
```

Comment:
This probably would not work if `substitute` is entered before `cost`

Reply:
Yep, idx needs to be reset since `argv_find` starts from `idx`

Reply:
Thanks for review and sorry for my miss! Done in last push.

Reply:
DEFPY really comes handy in cases like above.